test/cmd/newapp.sh

```diff
@@ -14,6 +14,48 @@ trap os::test::junit::reconcile_output EXIT
 os::util::environment::setup_time_vars
 
 os::test::junit::declare_suite_start "cmd/newapp"
+
+#
+# imagestream/tag creation and reuse
+#
+os::cmd::expect_success 'oc create -f examples/image-streams/image-streams-centos7.json -n openshift'
+os::cmd::expect_success 'oc delete imagestreamtag php:latest -n openshift'
+os::cmd::expect_success 'oc new-project test-imagestreams'
+os::cmd::try_until_failure 'oc get imagestreamtag php:latest -n openshift'
+
+# should fail due to missing php:latest tag
+os::cmd::expect_failure 'oc new-app --image-stream=openshift/php https://github.com/sclorg/cakephp-ex'
+
+# should succeed and create the php:latest tag in the current namespace
+os::cmd::expect_success 'oc new-app --docker-image=library/php https://github.com/sclorg/cakephp-ex --strategy=source'
+os::cmd::try_until_success 'oc get imagestreamtag php:latest -n test-imagestreams'
+os::cmd::expect_success 'oc create imagestreamtag php:latest -n openshift'
+
+# create a new tag for an existing imagestream in the current namespace
+os::cmd::expect_success 'oc create imagestreamtag perl:5.0'
+os::cmd::expect_success 'oc new-app --docker-image=library/perl https://github.com/sclorg/dancer-ex --strategy=source'
+os::cmd::try_until_success 'oc get imagestreamtag perl:latest -n test-imagestreams'
+
+# remove redundant imagestream tag before creating objects
+os::cmd::expect_success_and_text 'oc new-app  openshift/ruby-22-centos7 https://github.com/openshift/ruby-hello-world  --strategy=docker --loglevel=5' 'Removing duplicate tag from object list'
+
+# create imagestream in the correct namespace
+os::cmd::expect_success 'oc new-app --name=mytest --image-stream=mysql --env=MYSQL_USER=test --env=MYSQL_PASSWORD=redhat --env=MYSQL_DATABASE=testdb -l app=mytest'
+os::cmd::try_until_success 'oc get is mytest -n test-imagestreams'
```

Comment:
why would this create an imagestream?  you asked new-app to create an app based on an existing imagestream (named `mysql`, located in either this namespace or the openshift namespace).  I wouldn't expect it to ever create an imagestream though (you haven't pointed to an image, so what image would the imagestream reference?)

Reply:
I have based this test off of this bugzilla https://bugzilla.redhat.com/show_bug.cgi?id=1611462 
This command was trying to create the imagestream in the openshift namespace, now it creates it in the users namespace (and it does create an imagestream)